### ISSUE TITLE
Hostile Dragons Disappear Ocassionally

### ISSUE THREAD

**Issue by Xvareon** [Rebirth-of-the-Night/Ice_and_Fire#48]
- This is an ocassional bug that I have been able to reproduce on multiple ocassions.
-  To reproduce: fight any kind of dragon, sometimes they just disappear mid-fight. No trace, not in nearby chunks. Just poof.
- These are naturally spawned and encountered dragons. Nests and Caverns.
- I do not have this issue with original Ice and Fire
-  I am playing with other mods.
- No crashlogs since its not crashes
- The notable information I only have in my logs is "Client thread/Error: couldn't render entity, java.lang.NullPointerException:null"
- I am using forge 2860 (latest) for 1.12.2 and I am using 1.3.1 of inf: rotn
- I hope this issue gets fixed soon, I have a clue what's causing it and it may be the roost AI again.
- I get this bug more frequently on lightning dragons.

**Comment by Xvareon** [comment]
nvm the clue, I think its also a rendering issue

**Comment by democat3457** [comment]
can you send the whole log file, preferably the debug.log

**Comment by Xvareon** [comment]
Unfortunately, the logs have been buried among the other logs. This bug is also not easily reproduceable but rest assured it really is there.

**Comment by Xvareon** [comment]
Will send the logs if it happens again

**Comment by Xvareon** [comment]
![2023-12-10_12 45 50](https://github.com/Rebirth-of-the-Night/Ice_and_Fire/assets/91329439/1fbe08db-d157-4376-9324-070874bc3f1f)

This time, the dragons duplicated. Sometimes they disappear, sometimes they mutiply. Weird. This is the full log since I cannot find the debug log.
[Uploading latest.log…]()

**Comment by Xvareon** [comment]
[latest.log](https://github.com/Rebirth-of-the-Night/Ice_and_Fire/files/13626553/latest.log)

**Comment by Xvareon** [comment]
Happened again as of this time, a t5 fire dragon inside a cave. The logs don't provide anything.

**Comment by Xvareon** [comment]
[latest.log](https://github.com/Rebirth-of-the-Night/Ice_and_Fire/files/13626877/latest.log)

Can confirm its not the rendering error. For additional information, this happens mostly when dragons are knockedback, breaking blocks in the process while biting me, the player. Can't fully back up the scenario to consistently reproduce it.

**Comment by Xvareon** [comment]
Happened with t4 lightning dragon this time, while not breaking blocks but biting only

**Comment by Xvareon** [comment]
One culprit might also be the multipart system

**Comment by democat3457** [comment]
> Can confirm its not the rendering error.

yeah the rendering error seems to be from ancient spellcraft, possibly with entity culling

**Comment by Norfried** [comment]
Editing this config setting for Entity Culling might just do the trick in fixing this bug. Idk for certain, but it's worth a shot.
![EntityCullingDragonFix](https://github.com/user-attachments/assets/9ba10d9d-0ee0-480e-aa75-580ce9819b9f)
